src/Dapr.PluggableComponents/Components/IPluggableComponent.cs

```diff
@@ -0,0 +1,29 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.PluggableComponents.Components;
+
+/// <summary>
+/// Represents the root interface of all Dapr Pluggable Components.
+/// </summary>
+public interface IPluggableComponent
```

Comment:
Would this plus the next 2 interfaces be better served as an abstract base class? We don't need to change it, but it seems like some default methods for ping/features would be cleaner than having each of these as an interface. If you rely on the interfaces for some type checking, that's also OK.

Reply:
The interfaces *are* used, in part, in the service registration process.  Still, I'd argue that an abstract base class, while a perfectly valid approach, doesn't provide significant benefit over interfaces here as there is no significant common implementation or logic that it could provide.  The developer must still override and provide complete implementations for each that will vary widely based on the type of component.  Given that, I feel (simple) interfaces provide more clarity in what is expected from the developer.